batch/cogroup-operator/src/main/java/CoGroupSample.java

```diff
@@ -0,0 +1,95 @@
+/*
+ * Copyright (c) 2008-2017, Hazelcast, Inc. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import com.hazelcast.jet.DAG;
+import com.hazelcast.jet.Jet;
+import com.hazelcast.jet.JetInstance;
+import com.hazelcast.jet.Vertex;
+import com.hazelcast.jet.samples.cogroup.GeneratePersonsP;
+import com.hazelcast.jet.samples.cogroup.Person;
+
+import java.util.Arrays;
+
+import static com.hazelcast.jet.Edge.between;
+import static com.hazelcast.jet.Edge.from;
+import static com.hazelcast.jet.Processors.writeLogger;
+import static com.hazelcast.jet.samples.cogroup.Person.EMPLOYEES;
+import static com.hazelcast.jet.samples.cogroup.Person.STUDENTS;
+
+/**
+ * A sample demonstrating implementation of CoGroup operator. The operator
+ * joins two sources.
+ * <p>
+ * Also see comments in {@link CoGroupP}.
+ * <p>
+ * The DAG is as follows:
+ * <pre>{@code
+ *                 +-----------------+                 +-------------------+
+ *                 | Students source |                 | Employees source  |
+ *                 +-----------+-----+                 +----+--------------+
+ *                             |                            |
+ *            Person (student) |                            | Person (employee)
+ * (distributed, partitioned   |     +------------+         |  (distributed
+ *            priority edge)   +-----+  CoGroup   +---------+  partitioned edge)
+ *                                   +-----+------+
+ *                                         |
+ *                                         | Object[]{student, epmloyee}
+ *                                         |
+ *                                   +-----+------+
+ *                                   |    Sink    |
+ *                                   +------------+
+ * }</pre>
+ */
+public class CoGroupSample {
+
+    public static void main(String[] args) throws Exception {
+        System.setProperty("hazelcast.logging.type", "log4j");
+        JetInstance jet = Jet.newJetInstance();
+//        Jet.newJetInstance();
```

Comment:
I left it intentionally commented, because it gives ugly results. The `GeneratePersonsP` generates the same set of items on all instances. The results are then hard to interpret.